docs: add GUI program, requirements, and change Readme

Rizqidx opened the pull request:
This commit add GUI.py file as GUI program, assets folder include logo.ico as logo app, and requirements file. and also change the contents of Readme file that include overview and feature on GUI and installation guide.

Rizqidx committed:
feat: GUI program, requirements, and change Readme

This commit add GUI.py file, assets folder, and requirements file. and
also change the contents of Readme file that include overview and
feature on GUI and installation guide.

marinternet reviewed (changes requested):
Hello yanv, 

Great work for the GUI! Responsive enough through different resolutions and uploading images works.

A note for you is, next time, don't do all the lots of stuff in a single commit. Please break to several commits for example

1st commit: grid layouts ( without content ), so we can discuss the layout and responsiveness
2nd commit: contents for the placeholder, in this commit then, you can put the contents like upload image, predict, logs, etc. and then we can discuss the component placement and everything.

This ensures clean and step by step development, rather than you do all the stuff on the GUI and expects me to check every single thing in one go. 

Comments on the code

1. You are inputting all the features in one single code, look to make a separation between the UI and the logic of the code. This might works now, but when scaling ( when we later input the YOLO logic, segmentation logic, you will have a hard time maintaining the code and it will be like a 2000 line python code bomb!!

Suggestion: separate UI and logic with different files, and have it refer to the file with specific use. make a separate folder for UI and the Logic ( frequently the naming are 'core' or 'logic' ). This ensures easy and maintainable folder structure.

2. Implement a better error handling, look to use something like a message box to indicate the user what is wrong ( will be extremely important on later use )

Rizqidx committed:
refactor(gui): modularize application structure

this commit include split GUI.py into 4 file such as  main.py, left_panel.py,
middle_panel.py, and right_panel.py, and remove deprecated GUI.py.
